src/XRP/issued-currency-client.ts

```diff
@@ -94,6 +106,67 @@ export default class IssuedCurrencyClient {
     return trustLines
   }
 
+  /**
+   * Returns information about the total balances issued by a given account,
+   * optionally excluding amounts held by operational addresses.
+   * @see https://xrpl.org/issuing-and-operational-addresses.html
+   *
+   * @param account The account for which to retrieve balance information, encoded as an X-Address.
+   * @param hotwallet (Optional) An operational address to exclude from the balances issued, or an array of such addresses,
+   *                   encoded as X-Addresses.
+   * @see https://xrpaddress.info/
+   * @returns A GatewayBalances object containing information about an account's balances.
+   */
+  public async getGatewayBalances(
+    account: string,
+    hotwallet?: string | Array<string>,
+  ): Promise<GatewayBalances> {
+    // check issuing account for X-Address format
+    const classicAddress = XrpUtils.decodeXAddress(account)
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+
+    // check hotwallet addresses for X-Address format
+    const hotwalletArray: Array<string> = []
+    if (hotwallet !== undefined) {
+      if (typeof hotwallet == 'string') {
+        const hotwalletClassicAddress = XrpUtils.decodeXAddress(hotwallet)
+        if (!hotwalletClassicAddress) {
+          throw XrpError.xAddressRequired
+        }
+        hotwalletArray.push(hotwalletClassicAddress.address)
+      } else {
+        for (const address of hotwallet) {
+          const hotwalletClassicAddress = XrpUtils.decodeXAddress(address)
+          if (!hotwalletClassicAddress) {
+            throw XrpError.xAddressRequired
+          }
+          hotwalletArray.push(hotwalletClassicAddress.address)
+        }
+      }
+    }
+    // TODO: verify address format of hotwallet (and add test)
+    const gatewayBalancesResponse: GatewayBalancesResponse = await this.jsonNetworkClient.getGatewayBalances(
+      classicAddress.address,
+      hotwalletArray,
+    )
+
+    if (gatewayBalancesResponse.result.error) {
```

Comment:
i think this would be easier to reason about if you inverted it and returned early, ie

```
if (!gatewayBalancesResponse.result.error) {
  return new GatewayBalances(gatewayBalancesResponse)
}
```

Reply:
agreed, swapped